src/rustdoc_schema.graphql

```diff
@@ -694,6 +694,37 @@ type Constant implements Item & Importable & GlobalValue {
   docs: String
   attrs: [String!]!
   visibility_limit: String!
+  # properties for Constant
+  """
+  A type represented in the "raw" rustdoc JSON representation, see interface RawType.
+  Currently serialized as a serde_json String instead of a proper RawType
+  ```rust
+  //          	                	            	// rustdocs type field
+  const MIN   	:          usize	= 16       ;	// {"primitive":"usize"}
+  const LOG_AS	: &'static str  	= "batch"  ;	// {"borrowed_ref":{"lifetime":"'static","mutable":false,"type":{"primitive":"str"}}}
+  const YEAR  	:          Years	= Years(42);	// {"resolved_path":{"name":"Years","id":"0:3:1633","args":{"angle_bracketed":{"args":[],"bindings":[]}}}}
+  ```
+  """
+  type_: String
+  """
+  The expression/value of the constant, if any, as a Rust literal, expression, or `"_"`, as well as its literal flag. For example:
+  ```rust
+  //            	                	            	// is_literal	value      	expr
+  const MIN     	:          usize	= 16       ;	// true      	`"16usize"`	`16`
+  const MIN_SIZE	:          usize	= MIN      ;	// false     	`"16usize"`	"MIN", referring to the other constant's name
+  const LOG_AS  	: &'static str  	= "batch"  ;	// true      	None       	`"\"batch\""` including escaped quotes
+  const YEAR    	:          Years	= Years(42);	// false     	None       	"_"
+  const EXPR_2_2	:          i32  	= 2 + 2    ;	// false     	`"4i32"    	"_"
+  const FN_FIVE 	:          i32  	= five()   ;	// false     	`"5i32"    	"_"
```

Comment:
Let's try to make this representation a bit narrower, for example by reducing the spacing in the Rust code. Docs are often displayed in a side panel or on hover, where the panel can often be well under 100 chars. We don't have to squeeze every last char of width out (and the alignment here is very nice!) but we should avoid spending 10 spaces where 1-2 would do.

Reply:
should be better now

By the way, where do these docs go? Checked cargo docs and the crate docs site and don't see any schema docs

Reply:
They go in the "schema" pane of playgrounds — in this case, https://play.predr.ag/rustdoc which unfortunately at the moment has broken side scrolling because I suck at front-end engineering 😅 

![image](https://github.com/obi1kenobi/trustfall-rustdoc-adapter/assets/2348618/ce202c60-6ab3-45a8-b09c-0a8146adad4c)

They are also shown on hover in editors with a language server for GraphQL, just like rust-analyzer shows docs on hover for Rust code.

Reply:
Ok, so I've set up a local editor LSP and can see the comments live

But then within Rust codeblocks the single backticks markdown formatting is ignored (makes sense since this is regular rust comments now, and even doc comments aren't converted since this isn't done via cargo but LSP)
```
type Item;  // `"Item"` is the name of the associated type
```

so I'll remove them as they serve no purpose here?

And re. your previous ?, since there is no guarantee that all fields are present in an actual query, will go with your 2nd dupe option